holidays/financial/ny_stock_exchange.py

```diff
@@ -37,12 +37,12 @@ class NewYorkStockExchange(
     """New York Stock Exchange holidays.
 
     References:
-        * <https://www.nyse.com/publicdocs/nyse/regulation/nyse/NYSE_Rules.pdf>
-        * <https://www.nyse.com/markets/hours-calendars>
+        * <https://web.archive.org/web/20240118104341/https://www.nyse.com/publicdocs/nyse/regulation/nyse/NYSE_Rules.pdf>
+        * <https://web.archive.org/web/20250426144831/https://www.nyse.com/markets/hours-calendars>
```

Comment:
```suggestion
```

Reply:
>         * <https://web.archive.org/web/20240118104341/https://www.nyse.com/publicdocs/nyse/regulation/nyse/NYSE_Rules.pdf>

and

>         * <https://web.archive.org/web/20250426144831/https://www.nyse.com/markets/hours-calendars>

are two different links AFAIK

Reply:
We have https://www.nyse.com/markets/hours-calendars in "Historical data" below.